.github/workflows/python-package.yml

```
@@ -0,0 +1,29 @@
+name: Python package
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+
+jobs:
+  build:
+
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: ["3.9", "3.10", "3.11"]
```

Are older versions not supported by Github Actions? Previously we were testing on Python 2 too.

---

Looks like https://github.com/actions/python-versions does not support Python 2. I picked three versions to reduce tasks.

---

It looks like 3.5 is the oldest version we can test with. Let's assume that it isn't broken in intervening versions. Also, we can use 3.x to automatically pick up new stable versions so we will notice if a new version is broken.
```suggestion
        python-version: ["3.5", "3.10", "3.x"]
```

---

Python 3.7 and lower reached end-of-life. Tests also fail on the latest version (3.12). We should fix tests but it should be done in a separate PR?

---

For the time being we also support Python 2, which is also end-of-life, so our support decisions are not necessarily tied to Python's.

Let's stick with this for now and fix 3.12 in a separate PR.

---

Makes sense. Replaced 3.8 with 3.5. Can you squash & merge? It seems that I don't have the permission.

---

Hmm 3.5 didn't work. Changed to 3.7.